File: iOSApp/SplitTab/Guest.m
//
//  Guest.m
//  SplitTab
//
//  Created by Miranda Strand on 7/16/14.
//  Copyright (c) 2014 FUZZYturtles. All rights reserved.
//

#import "Guest.h"

@implementation Guest {
    //private instance variables go here
}

@synthesize name = _name;
@synthesize owed = _owed;

-(id)initWithName:(NSString *)newName :(int)twenties :(int)tens :(int)fives :(int)ones
{
    self = [super init];
    if (self) {
        _name = newName;
        _billsPaid = [[NSMutableArray alloc] init];
        _change = [NSMutableArray arrayWithObjects:@0, @0, @0, @0, nil];
        [_billsPaid addObject:[NSNumber numberWithInt:twenties]];
        [_billsPaid addObject:[NSNumber numberWithInt:tens]];
        [_billsPaid addObject:[NSNumber numberWithInt:fives]];
        [_billsPaid addObject:[NSNumber numberWithInt:ones]];
    }
    return self;
}

-(void)setOwed:(int)owed
{
    _owed = owed;
}

-(int)twenties
{
    return [_billsPaid[0] intValue];
}

-(int)tens
{
    return [_billsPaid[1] intValue];
}

-(int)fives
{
    return [_billsPaid[2] intValue];
}

-(int)ones
{
    return [_billsPaid[3] intValue];
}

@end

File: iOSApp/algorithm/src/Guest.java

public class Guest {
	//fields
	private String name;
	public int [] bills; //public for now out of laziness
	private int owed;
	
	public Guest(String newname, int twenties, int tens, int fives, int ones) {
		name = newname;
		bills = new int[4];
		bills[0] = twenties;
		bills[1] = tens;
		bills[2] = fives;
		bills[3] = ones;
	}
	
	public int twenties() {
		return bills[0];
	}
	
	public int tens() {
		return bills[1];
	}
	
	public int fives() {
		return bills[2];
	}
	
	public int ones() {
		return bills[3];
	}
	
	public void setOwed(int cost) {
		owed = cost;
	}
	
	public int getOwed() {
		return owed;
	}
	
	public String getName() {
		return name;
	}
	
	public String toString() {
		String str = "Name: " + name + " Twenties: " + bills[0] + " Tens: " + bills[1] + " Fives: " + bills[2] + " Ones: " + bills[3];
		return str;
	}
}
